test/e2e/nodeconditions.go

```diff
@@ -0,0 +1,249 @@
+				value: strconv.Itoa(time.Now().Nanosecond()),
+			}
+			createPod(c, ns, id, podCPU)
+			podIds = append(podIds, id)
+		}
+		defer deletePods(c, ns)
+
+		By("Waiting for all but one pods to be scheduled")
+		// Wait for all pods except the last one to be running
+		for _, id := range podIds[:len(podIds)-1] {
+			expectNoError(framework.WaitForPodRunning(id.name))
+			pods, err := podClient.List(labels.SelectorFromSet(labels.Set(map[string]string{"time": id.value})), fields.Everything())
+			expectNoError(err)
+			for _, pod := range pods.Items {
+				Expect(pod.Spec.NodeName).ToNot(Equal(""))
+			}
+		}
+
+		By(fmt.Sprintf("Finding a failed scheduler event for pod %s", podIds[len(podIds)-1].name))
+		schedEvents, err := c.Events(ns).List(
+			labels.Everything(),
+			fields.Set{
+				"involvedObject.kind":      "Pod",
+				"involvedObject.name":      podIds[len(podIds)-1].name,
+				"involvedObject.namespace": ns,
+				"source":                   "scheduler",
+				"reason":                   "FailedScheduling",
+			}.AsSelector())
+		expectNoError(err)
+
+		Expect(schedEvents.Items).ToNot(BeEmpty())
+	})
+})
+
+func createPod(c *client.Client, ns string, id podId, milliCPU int64) {
+	podClient := c.Pods(ns)
+
+	pod := &api.Pod{
+		ObjectMeta: api.ObjectMeta{
+			Name: id.name,
+			Labels: map[string]string{
+				"name": "foo",
+				"time": id.value,
+			},
+		},
+		Spec: api.PodSpec{
+			Containers: []api.Container{
+				{
+					Name:  "nginx",
+					Image: "beta.gcr.io/google_containers/pause:2.0",
+					Resources: api.ResourceRequirements{
+						Requests: api.ResourceList{
+							// Request enough CPU to fit only two pods on a given node.
+							api.ResourceCPU:     *resource.NewMilliQuantity(milliCPU, resource.DecimalSI),
+							api.ResourceStorage: *resource.NewQuantity(50*gb, resource.DecimalSI),
+						},
+					},
+				},
+			},
+		},
+	}
+
+	_, err := podClient.Create(pod)
+	expectNoError(err)
+}
+
+func deletePods(c *client.Client, ns string) {
+	By("Removing all pods in namespace " + ns)
+	pods, err := c.Pods(ns).List(labels.Everything(), fields.Everything())
+	expectNoError(err)
+	opt := api.NewDeleteOptions(0)
+	for _, p := range pods.Items {
+		expectNoError(c.Pods(ns).Delete(p.ObjectMeta.Name, opt))
+	}
+}
+
+func availSize(c *client.Client, node api.Node) (uint64, error) {
+	statsResource := fmt.Sprintf("api/v1/proxy/nodes/%s/stats/", node.Name)
+	Logf("Querying stats from node %s using url %s", node.Name, statsResource)
+	res, err := c.Get().AbsPath(statsResource).Timeout(timeout).Do().Raw()
+	if err != nil {
+		return 0, fmt.Errorf("error querying cAdvisor API: %v", err)
+	}
+	ci := cadvisorapi.ContainerInfo{}
+	err = json.Unmarshal(res, &ci)
+	if err != nil {
+		return 0, fmt.Errorf("couldn't unmarshal container info: %v", err)
+	}
+	return ci.Stats[len(ci.Stats)-1].Filesystem[0].Available, nil
+}
+
+func fillDiskSpace(c *client.Client) {
+	// Get all nodes, and kick off the test on each.
```

Comment:
I think you just want to say "fill up the disk on every node except the one with index 0."
Also I think we've been putting function-level comments above the function.